lib/stdio/generator.js

```diff
@@ -0,0 +1,86 @@
+import {Duplex, Readable, PassThrough, getDefaultHighWaterMark} from 'node:stream';
+import {callbackify} from 'node:util';
+import {setTimeout} from 'node:timers/promises';
+
+// Generators can be used to transform/filter standard streams.
+// Generators have a simple syntax, yet allows all of the following:
+//  - Sharing state between chunks, by using logic before the `for` loop
+//  - Flushing logic, by using logic after the `for` loop
+//  - Asynchronous logic
+//  - Emitting multiple chunks from a single source chunk, even if spaced in time, by using multiple `yield`
+//  - Filtering, by using no `yield`
+// Therefore, there is no need to allow Node.js or web transform streams.
+// The `highWaterMark` is kept as the default value, since this is what `childProcess.std*` uses.
+// We ensure `objectMode` is `false` for better buffering.
+// Chunks are currently processed serially. We could add a `concurrency` option to parallelize in the future.
```

Comment:
I would prefer a `/* */` comment as it makes it easier to edit it.

Reply:
Done in daf5b9c4185a473a74edf18e35c20d77b7410434